src/box_utils.cpp

```diff
@@ -214,6 +228,15 @@ Box padBox(const Box& box, double padding)
   return box_padded;
 }
 
+std::vector<Box> padBoxArray(const std::vector<Box>& box_array, double padding)
+{
+  std::vector<Box> box_array_padded(box_array.size());
+  std::transform(box_array.cbegin(), box_array.cend(), box_array_padded.begin(),
+                 [&padding](const Box box) -> Box { return padBox(box, padding); });
```

Comment:
It's better to use std::back_inserter in this case. Can you understand why?

```suggestion
  std::vector<Box> box_array_padded;
  box_array_padded.reserve(box_array.size());
  std::transform(box_array.cbegin(), box_array.cend(), std::back_inserter(box_array_padded),
                 [&padding](const Box box) -> Box { return padBox(box, padding); });
```